### DISCUSSION TITLE
Reporting that EOI is also possible?

### DISCUSSION THREAD

**Discussion by carueda** [zesterer/chumsky#540] [Q&A] [answered]
(First off, great crate!)

With nano_rust.rs as reference, I wonder how to make the parser indicate that EOI is also possible when encountering some unexpected stuff right after valid input.

Suppose I add "HMM" at the end of `sample.nrs`:

```
cargo run --features="label" --example nano_rust -- examples/sample.nrs
```

```
    ╭─[examples/sample.nrs:39:1]
    │
 39 │ HMM
    │ ─┬─
    │  ╰─── found 'HMM' expected 'fn'
────╯
```

Thanks.

**Comment by carueda** [comment]
A similar exercise with the json.rs example, and with this input:

```json
"foo"
HMM
```

Gives:

```
Error: found 'H' expected something else
   ╭─[<unknown>:2:1]
   │
 2 │ HMM
   │ ┬
   │ ╰── found 'H' expected something else
───╯
```

but "expected end of input" or similar would be more desirable.

(Adding `.then_ignore(end())` to the top-level parser did not change that report behavior.)

**Comment by zesterer** [comment, marked as answer]
Ah, it turns out that this was a minor bug in the implementation of `end`, likely caused by an internal change made a few months ago. I've just fixed this in 36f8e8b and all should now be good :)

![image](https://github.com/zesterer/chumsky/assets/3268430/8add4992-c237-4de5-a94f-86ce507f6da3)

**Comment by carueda** [comment]
Thanks!
